project/Component/NefModels/ProgressReport.swift

```diff
@@ -0,0 +1,52 @@
+import BowEffects
+
+public protocol ProgressReport {
+    func notify<E: Error, A: CustomProgressDescription>(_ event: ProgressEvent<A>) -> IO<E, Void>
+}
+
+public extension ProgressReport {
+    func oneShot<E: Error, A: CustomProgressDescription>(_ step: A) -> IO<E, Void> {
+        self.notify(
+            ProgressEvent(step: step,
+                          status: .oneShot))
+    }
+    
+    func inProgress<E: Error, A: CustomProgressDescription>(_ step: A) -> IO<E, Void> {
+        self.notify(
+            ProgressEvent(step: step,
+                          status: .inProgress))
+    }
+    
+    func succeeded<E: Error, A: CustomProgressDescription>(_ step: A, info: String = "") -> IO<E, Void> {
+        self.notify(
+            ProgressEvent(step: step,
+                          status: .successful(info: info)))
+    }
+    
+    func failed<E: Error, A: CustomProgressDescription>(_ step: A, _ error: E, info: String = "") -> IO<E, Void> {
+        self.notify(
+            ProgressEvent(step: step,
+                          status: .failed(error, info: info)))
+    }
+}
+
+public enum ProgressEventStatus {
+    case oneShot
+    case inProgress
+    case successful(info: String)
```

Comment:
Ok, I see, we are mixing the success of an event, with the finish (status) task. You do not need add `info` in successful or failed status, it comes from `oneShot` or `inProgress` cases.